### PR TITLE
Teach CLI how to execute resource transfers.

### PR DESCRIPTION
**Pull request by jelmersnoeck**
Once we've deployed the change to the marketplace, we can merge this in.

### CODE REVIEW

File: cmd/transfer.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+		idx, _, err := prompts.SelectResource(resources, projects, resourceName)
+		if err != nil {
+			return prompts.HandleSelectError(err, "Could not select Resource")
+		}
+		resource = resources[idx]
+	}
+
+	fmt.Println(newOwner)
+	newOwnerID, err := newOwnerID(ctx, newOwner, client.Identity)
+	fmt.Println("fooooo", err)
```
Comment (luizbranco) [inline cmd/transfer.go]: debug? 😆

File: cmd/transfer.go [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+	if err != nil {
+		return err
+	}
+
+	dontWait := cliCtx.Bool("no-wait")
+	if err := transferResource(ctx, client, userID, teamID, resource.ID, *newOwnerID, dontWait); err != nil {
```
Comment (luizbranco) [inline cmd/transfer.go]: Show success msg?

File: cmd/transfer.go [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+		}
+	} else {
+		idx, _, err := prompts.SelectResource(resources, projects, resourceName)
+		if err != nil {
+			return prompts.HandleSelectError(err, "Could not select Resource")
+		}
+		resource = resources[idx]
+	}
+
+	fmt.Println(newOwner)
```
Comment (luizbranco) [inline cmd/transfer.go]: This can be removed

File: cmd/transfer.go [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+func newOwnerID(ctx context.Context, newOwner string, ic *iClient.Identity) (*manifold.ID, error) {
+	teams, err := clients.FetchTeams(ctx, ic)
+	if err != nil {
+		return nil, err
+	}
+
+	isEmail := govalidator.IsEmail(newOwner)
+	for _, t := range teams {
+		fmt.Println(string(t.Body.Label), newOwner)
+		fmt.Println(string(t.Body.Label) == newOwner)
```
Comment (luizbranco) [inline cmd/transfer.go]: Debug?

File: cmd/transfer.go [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+
+	dontWait := cliCtx.Bool("no-wait")
+	if err := transferResource(ctx, client, userID, teamID, resource.ID, *newOwnerID, dontWait); err != nil {
+	}
+
+	return nil
+}
+
+func newOwnerID(ctx context.Context, newOwner string, ic *iClient.Identity) (*manifold.ID, error) {
+	teams, err := clients.FetchTeams(ctx, ic)
```
Comment (luizbranco) [inline cmd/transfer.go]: If the newOwner is not a valid email format, we could just fetch a single team. I'm going to create an issue to add a label filter to GET /teams.

File: cmd/transfer.go [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	isEmail := govalidator.IsEmail(newOwner)
+	for _, t := range teams {
+		fmt.Println(string(t.Body.Label), newOwner)
+		fmt.Println(string(t.Body.Label) == newOwner)
+		if isEmail {
```
Comment (luizbranco) [inline cmd/transfer.go]: Does the user needs to be a member of a shared team? Can't I just transfer to any registered user?

**Comment by jelmersnoeck** [comment]
@luizbranco I've added a better description with regards to the team relationship. Also added a proper success message.

**Pull request merged**